rpc/grpc/client/block_results_service.go

```diff
@@ -0,0 +1,68 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	"github.com/cosmos/gogoproto/grpc"
+
+	brs "github.com/cometbft/cometbft/proto/tendermint/services/block_results/v1"
+)
+
+type BlockResults struct {
+	Height                int64                     `json:"height"`
+	TxsResults            []*abci.ExecTxResult      `json:"txs_results"`
+	FinalizeBlockEvents   []*abci.Event             `json:"finalize_block_events"`
+	ValidatorUpdates      []*abci.ValidatorUpdate   `json:"validator_updates"`
+	ConsensusParamUpdates *cmtproto.ConsensusParams `json:"consensus_param_updates"`
+	AppHash               []byte                    `json:"app_hash"`
+}
+
+// BlockResultsServiceClient provides the block results of a given height (or latest if none provided).
+type BlockResultsServiceClient interface {
+	GetBlockResults(ctx context.Context, height *int64) (*BlockResults, error)
```

Comment:
Would it not be better to have `height` be of type `int64`, similar to the block service's client interface?

Using `int64` seems more ergonomic than having to supply pointers to this function.

Reply:
See [the latest commit](https://github.com/cometbft/cometbft/pull/1168/commits/14372cb1c560f77d96f4da3cd73cdacd86c359d3) with an update to provide an additional endpoint specifically to retrieve `BlockResults` of the latest committed height.